components/hab/habitat/plan.sh

```diff
@@ -60,15 +56,12 @@ do_prepare() {
   export rustc_target="x86_64-unknown-linux-musl"
   build_line "Setting rustc_target=$rustc_target"
 
+  # TODO (DM): `core/openssl-musl` includes the `core/zlib-musl` package even though
```

Comment:
This is likely due to the kernel2 package not having been rebuilt  since the `no-zlib` change was introduced.  This _may_ cause us some problems between `x86_64-linux` and `x86_86-linux-kernel2`.   We should work with the core-plans maintainers to get a build of openssl-musl for `x86_86-linux-kernel2` to correct this, and ensure we're using the latest version of openssl.

Reply:
Good catch, thanks! This looks like exactly what is happening and does cause issues between the platforms. I opened a new PR (#7710) that builds on top of this PR, but it also vendors openssl fixing this problem. If #7710 gets approved I can merge that PR and close this one.